pkg/services/ngalert/metrics/remote_alertmanager.go

```diff
@@ -80,5 +81,12 @@ func NewRemoteAlertmanagerMetrics(r prometheus.Registerer) *RemoteAlertmanager {
 			Name:      "remote_alertmanager_last_state_sync_timestamp_seconds",
 			Help:      "Timestamp of the last successful state sync to the remote Alertmanager in seconds.",
 		}),
+		ConfigSizeBytes: promauto.NewHistogram(prometheus.HistogramOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "grafana_alerting_alertmanager_config_size_bytes",
+			Help:      "The size of the grafana alertmanager configuration file in bytes",
+			Buckets:   prometheus.ExponentialBuckets(512, 2, 10),
+		}),
```

Comment:
This metric looks like a fit for the [Grafana Alertmanager metrics](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/metrics/alertmanager.go), not the remote AM ones.

We can use a gauge to expose the size of the AM config when we apply it, that way we'll know about it even if the `remote.alertmanager.enabled` is set to `false`.

[This method](https://github.com/grafana/grafana/blob/9b2c2be82d643632cbc48112595c068e89851c0f/pkg/services/ngalert/notifier/alertmanager.go#L286-L298) looks interesting, it's called every time we apply a config in the internal AM, we could update the gauge there.

Reply:
Good points! Changed now 👍